core/services/pipeline/task.eth_call.go

```diff
@@ -115,23 +118,29 @@ func (t *ETHCallTask) Run(ctx context.Context, lggr logger.Logger, vars Vars, in
 		}
 	}
 
-	call := ethereum.CallMsg{
-		To:        (*common.Address)(&contractAddr),
-		From:      (common.Address)(from),
-		Data:      []byte(data),
-		Gas:       uint64(selectedGas),
-		GasPrice:  gasPrice.BigInt(),
-		GasTipCap: gasTipCap.BigInt(),
-		GasFeeCap: gasFeeCap.BigInt(),
+	args := map[string]interface{}{
+		"from":                 common.Address(from),
+		"to":                   (*common.Address)(&contractAddr),
+		"input":                hexutil.Bytes([]byte(data)),
+		"gas":                  hexutil.Uint64(uint64(selectedGas)),
+		"gasPrice":             (*hexutil.Big)(gasPrice.BigInt()),
+		"maxFeePerGas":         (*hexutil.Big)(gasFeeCap.BigInt()),
+		"maxPriorityFeePerGas": (*hexutil.Big)(gasTipCap.BigInt()),
 	}
 
-	lggr = lggr.With("gas", call.Gas).
-		With("gasPrice", call.GasPrice).
-		With("gasTipCap", call.GasTipCap).
-		With("gasFeeCap", call.GasFeeCap)
+	selectedBlock, err := selectBlock(string(block))
+	if err != nil {
+		return Result{Error: err}, runInfo
+	}
+
+	lggr = lggr.With("gas", uint64(selectedGas)).
+		With("gasPrice", gasPrice.BigInt()).
+		With("gasTipCap", gasTipCap.BigInt()).
+		With("gasFeeCap", gasFeeCap.BigInt())
 
 	start := time.Now()
-	resp, err := chain.Client().CallContract(ctx, call, nil)
+	var resp hexutil.Bytes
+	err = chain.Client().CallContext(ctx, &resp, "eth_call", args, selectedBlock)
```

Comment:
I see that geth also implements a `pendingCallContract` function: https://github.com/ethereum/go-ethereum/blob/master/ethclient/ethclient.go#L541

Would this also be useful to explore?

Reply:
Admittedly, it seems like EstimateGas doesn't have the same pending option natively in geth.

Reply:
nice find. I think this will minimize the changes I need to make in this PR. Will try it out